VideoLocker/src/main/java/org/edx/mobile/module/db/impl/DbHelper.java

```diff
@@ -109,12 +116,60 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
+                                DbStructure.Column.FILEPATH}, null, null, null, null, null, null);
+                if (cursor != null) {
+                    final File appExternalDir = mContext.getExternalFilesDir(null).getParentFile();
+                    try {
+                        while (cursor.moveToNext()) {
+                            final int id = cursor.getInt(0);
+                            final String username = cursor.getString(1);
+                            final String filePath = cursor.getString(2);
+                            final String encryptedUsername = Sha1Util.SHA1(username);
+                            final String newFilePath = filePath.replace(
+                                    appExternalDir.getAbsolutePath() + "/" + username,
+                                    appExternalDir.getAbsolutePath() + "/"
+                                            + AppConstants.Folders.VIDEOS + "/"
+                                            + encryptedUsername);
+
+                            // First update the folder names
+                            final File previousFolder = new File(appExternalDir, username);
+                            if (previousFolder.exists()) {
+                                final File newFolder = new File(appExternalDir,
+                                        AppConstants.Folders.VIDEOS + "/" + encryptedUsername);
+                                if (!newFolder.exists()) newFolder.mkdirs();
+                                previousFolder.renameTo(newFolder);
+                            }
+
+                            // Then update the database rows
+                            final ContentValues updatedValues = new ContentValues();
+                            updatedValues.put(DbStructure.Column.USERNAME, encryptedUsername);
+                            updatedValues.put(DbStructure.Column.FILEPATH, newFilePath);
+                            db.update(DbStructure.Table.DOWNLOADS, updatedValues,
+                                    DbStructure.Column.ID + "=" + id, null);
+                        }
+                    } finally {
+                        cursor.close();
+                    }
+                    // Now migrate the subtitles folder
+                    final File previousSrtFolder = new File(appExternalDir, "srtFolder");
+                    if (previousSrtFolder.exists()) {
+                        final File newSrtFolder = new File(appExternalDir,
+                                AppConstants.Folders.VIDEOS + "/" + AppConstants.Folders.SUBTITLES);
+                        if (!newSrtFolder.exists()) newSrtFolder.mkdirs();
+                        previousSrtFolder.renameTo(newSrtFolder);
+                    }
+                }
+            }
+
             logger.debug("Database upgraded from " + oldVersion + " to " + newVersion);
         }catch(Exception e){
             logger.error(e);
         }
```

Comment:
We definitely should not be discarding the exception. If we do, the app will consider the upgrade successful even if it fails, and will never try to upgrade it again for this database version.